src/main/java/world/bentobox/bentobox/database/objects/Island.java

```diff
@@ -56,7 +58,7 @@
 public class Island implements DataObject, MetaDataAble {
 
     @Expose
-    private boolean primary;
+    private Set<UUID> primaries = new HashSet<>();
```

Comment:
This fixes a bug where the primary island flag was not specific to users and instead was global.